sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -2381,6 +2381,29 @@ def my_pipeline(sample_input1: bool = True,
         # test reloaded comments
         self.assertIn(predicted_comment, reloaded_yaml_content)
 
+    def test_comment_with_newline_chars(self):
+
+        @dsl.component
+        def foo(n: str = '\n', r: str = '\r'):
+            ...
```

Comment:
nit: Though compiler don't look into the function body, but I'd suggest at least keep it a valid Python syntax, e.g.: use `pass` instead of `...`.

Reply:
Done.